app/Helpers/RemoteData.php

```diff
@@ -43,42 +45,47 @@ public static function sync()
         $users = collect($response->getBody()['value']);
 
         $result = array(
-            "total" => Pointer::count(),
             "successful" => 0,
             "failed"=> 0
         );
 
-        # TODO: get only pointers with category "Person"
-        Pointer::all()->each(function (Pointer $pointer) use ($users, &$result) {
+        foreach ($building->levels as $level) {
 
-            $filtered = $users->filter(function($value) use ($pointer){
+            foreach (Level::find($level->id)->pointers as $pointer) {
 
-                if (strtolower($value['userPrincipalName']) == strtolower($pointer->meta)) {
-                    return true;
-                }
-            });
+                # syncing data only for people
+                if ($pointer->category->name == 'Person') {
+
+                    $filtered = $users->filter(function($value) use ($pointer){
 
-            $match = $filtered->first();
+                        if (strtolower($value['userPrincipalName']) == strtolower($pointer->meta)) {
+                            return true;
+                        }
+                    });
 
-            if ($match) {
+                    $match = $filtered->first();
 
-                Log::info("Found user: {$match['userPrincipalName']}.");
+                    if ($match) {
 
-                $pointer->name = "{$match['givenName']} {$match['surname']}";
-                $pointer->description = "Job Title: {$match['jobTitle']} <br> Department: {$match['department']}";
-                $pointer->save();
+                        Log::info("Found a user: {$match['userPrincipalName']}.");
 
-                $result['successful']++;
+                        $pointer->name = "{$match['givenName']} {$match['surname']}";
+                        $pointer->description = "Job Title: {$match['jobTitle']} <br> Department: {$match['department']}";
+                        $pointer->save();
 
-            } else {
+                        $result['successful']++;
 
-                Log::critical("Couldn't find user with email: {$pointer->meta}.");
```

Comment:
never use critical!!! use log::error instead please.

Reply:
Critical level should be used in more extreme or critical conditions, in this case the app can keep working normally, this error is raised for 1 item that not compromise the functional integrity of the app so imho this should be error or even warning level.

Check https://tools.ietf.org/html/rfc5424#page-11.

Reply:
fixed